### PR TITLE
read number of channels from Geometry

### PR DESCRIPTION
**Pull request by AllaMaevskaya**
Temporary solution for O2-2380 "Mem corruption in MC-AOD production"

### CODE REVIEW

File: Detectors/AOD/src/AODProducerWorkflowSpec.cxx [hunk 1]
```diff
@@ -606,21 +607,24 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
...
     }
-    float aAmplitudesA[96];
-    float aAmplitudesC[112];
-    for (int i = 0; i < 96; i++) {
-      aAmplitudesA[i] = truncateFloatFraction(vAmplitudes[i], mT0Amplitude);
-    }
-    for (int i = 0; i < 112; i++) {
-      aAmplitudesC[i] = truncateFloatFraction(vAmplitudes[i + 96], mT0Amplitude);
+    float aAmplitudesA[nFT0ChannelsAside];
+    float aAmplitudesC[133];
```
Comment (shahor02) [inline Detectors/AOD/src/AODProducerWorkflowSpec.cxx:621]: In this way you are introducing a dummy channels. Is not it better to introduce at once a proper mapping?
I thought something like 
````
int HW2SW(int id) const { return id < 200 ? id : (id>203 ? id - 4 : -1); }
int SW2HW(int id) const { return id<200 ? id : id + 4; }
````
should be enough.
Anyway, if it is more complex, I hope this solution is indeed temporary and will be overridden by a proper mapping.

Comment (jotwinow) [reply Detectors/AOD/src/AODProducerWorkflowSpec.cxx:621]: Alla, I would go for the general solution.

Comment (AllaMaevskaya) [reply Detectors/AOD/src/AODProducerWorkflowSpec.cxx:621]: General solution takes time

**Review by shahor02 — APPROVED** [review]
OK, merging this, but please, don't postpone correct solution.

**Pull request merged**